Spawn new alacritty process in CWD on macOS (#4299)
alacritty/alacritty: CHANGELOG.md

### Changed

- Cursors are now inverted when their fixed color is similar to the cell's background

> - Use working directory of the terminal foreground process instead of shell for SpawnNewInstance action

Reviewer: ```suggestion
- Use the working directory of the terminal foreground process, instead of the shell's working
    directory, for the `SpawnNewInstance` action
```

- Fallback to normal underline for unsupported underline types in `CSI 4 : ? m` escapes
- The user's background color is now used as the foreground for the render timer
- Use yellow/red from the config for error and warning messages instead of fixed colors
- Existing CLI parameters are now passed to instances spawned using `SpawnNewInstance`
- Wayland's Client side decorations now use the search bar colors
- Reduce memory usage by up to at least 30% with a full scrollback buffer
- The number of zerowidth characters per cell is no longer limited to 5

> - `SpawnNewInstance` is now using working directory of the terminal foreground process on macOS

Reviewer: ```suggestion
 - `SpawnNewInstance` is now using the working directory of the terminal foreground process on macOS
```

### Changed

- Cursors are now inverted when their fixed color is similar to the cell's background
- Fallback to normal underline for unsupported underline types in `CSI 4 : ? m` escapes
- The user's background color is now used as the foreground for the render timer
- Use yellow/red from the config for error and warning messages instead of fixed colors
- Existing CLI parameters are now passed to instances spawned using `SpawnNewInstance`
- Wayland's Client side decorations now use the search bar colors
- Reduce memory usage by up to at least 30% with a full scrollback buffer
- The number of zerowidth characters per cell is no longer limited to 5

> - `SpawnNewInstance` is now using working directory of the terminal foreground process on macOS/Linux

Reviewer: Adding the /Linux to the end made things go beyond our 100 character limit per line by 1.

It's also not entirely correct, since it applies to BSD too.

- Use yellow/red from the config for error and warning messages instead of fixed colors
- Existing CLI parameters are now passed to instances spawned using `SpawnNewInstance`
- Wayland's Client side decorations now use the search bar colors

> - `SpawnNewInstance` now starts in the same working directory on macOS

Reviewer 2: This still has strange message, I'd suggest to go through the previous comments wrt that entry, and update it.

- Wayland's Client side decorations now use the search bar colors
- Reduce memory usage by up to at least 30% with a full scrollback buffer
- The number of zerowidth characters per cell is no longer limited to 5

> - `SpawnNewInstance` is now using working directory of active process on macOS

Reviewer: ```suggestion
- `SpawnNewInstance` is now using the working directory of the terminal foreground process on macOS
```

"Active process" is honestly kind of a poor name for it. I know that the other changelog entry uses this wording already but that's just a bit misleading. The `terminal` is kinda optional, but I think it's a useful addition.

I'm open to changing the other changelog entry too.

- `Add` and `Subtract` keys are now named `NumpadAdd` and `NumpadSubtract` respectively
- Feature checking when cross compiling between different operating systems
- Crash when writing to the clipboard fails on Wayland

> - `SpawnNewInstance` will now remain in the same working directory on macOS

Reviewer 2: I don't really like the way it's written here.

Maybe something like
```
`SpawnNewInstance` not using working directory of active process on macOS
```

Reviewer: The way you rephrased it is a bugfix, which it isn't. This should actually be in the changed or added section.

Besides that, "will now remain" is a bit misleading, since it doesn't really remain anywhere. It changes into that directory.

Author: I put the change log entry in the "Fixed" section because it's clearly stated behaviors wasn't being respected in from the configuration:

https://github.com/alacritty/alacritty/blob/80688e082d13197b6661cfb6cfa60734034270cf/alacritty.yml#L375-L379

That said, I'll move this to the changed section, since that's applicable as well, and the remaining open issue (#1979) is marked as an enhancement.

Reviewer: Thanks for the clarification, but at that point I'd say it's more of a bug in our documentation than anything else. This definitely was planned to work exactly the way it does right now on macOS. Generally unless it's actually something that should be behaving differently, I wouldn't consider it a bug.

Having things fall in the `Fixed` section is also usually just the "goto" place for everything that isn't a new feature. So if there's any reason to move it somewhere else it's probably a good idea to do so.
